src/sql/platform/connection/common/connectionProfile.ts

```diff
@@ -65,10 +66,17 @@ export class ConnectionProfile extends ProviderConnectionInfo implements interfa
 				let capabilities = this.capabilitiesService.getCapabilities(this.providerName);
 				if (capabilities && capabilities.connection && capabilities.connection.connectionOptions) {
 					const options = capabilities.connection.connectionOptions;
+					// MSSQL Provider doesn't treat appName as special type anymore.
 					let appNameOption = options.find(option => option.specialValueType === interfaces.ConnectionOptionSpecialType.appName);
 					if (appNameOption) {
 						let appNameKey = appNameOption.name;
 						this.options[appNameKey] = Constants.applicationName;
+					} else if (this.providerName === Constants.mssqlProviderName) {
```

Comment:
I know.. not the best way to do this.. but curious to know is there a way to get this done by provider? Or should this be done for all providers instead? Not sure how many providers support appname, and if they also would receive the value from advanced options by making it non-special type; the appname option name wouldn't necessarily match..